pkg/controller/authenticationservice/standard.go

```
@@ -0,0 +1,216 @@
+)
+
+func applyStandardAuthServiceDefaults(ctx context.Context, client client.Client, scheme *runtime.Scheme, authservice *adminv1beta1.AuthenticationService) error {
+	if authservice.Spec.Standard == nil {
+		authservice.Spec.Standard = &adminv1beta1.AuthenticationServiceSpecStandard{}
+	}
+	if authservice.Spec.Standard.Storage == nil {
+		authservice.Spec.Standard.Storage = &adminv1beta1.AuthenticationServiceSpecStandardStorage{
+			Type: adminv1beta1.Ephemeral,
+		}
+	}
+	if authservice.Spec.Standard.CertificateSecret == nil {
+		authservice.Spec.Standard.CertificateSecret = &corev1.SecretReference{
+			Name: "standard-authservice-cert",
+		}
+	}
+	if authservice.Spec.Standard.CredentialsSecret == nil {
+		authservice.Spec.Standard.CredentialsSecret = &corev1.SecretReference{
+			Name: "keycloak-credentials",
+		}
+		secret := &corev1.Secret{}
+		err := client.Get(ctx, types.NamespacedName{Namespace: authservice.Namespace, Name: "keycloak-credentials"}, secret)
+		if err != nil && errors.IsNotFound(err) {
+			secret.ObjectMeta = metav1.ObjectMeta{Namespace: authservice.Namespace, Name: "keycloak-credentials"}
+			install.ApplyDefaultLabels(&secret.ObjectMeta, "standard-authservice", "keycloak-credentials")
+			secret.StringData = make(map[string]string)
+			secret.StringData["admin.username"] = "admin"
+			secret.StringData["admin.password"] = util.GenerateAlphaNumeric(32)
+			if err := controllerutil.SetControllerReference(authservice, secret, scheme); err != nil {
+				return err
+			}
+			err := client.Create(ctx, secret)
+			if err != nil {
+				return err
+			}
+		}
+		return err
+	}
+	return nil
+}
+
+func applyStandardAuthServiceDeployment(authservice *adminv1beta1.AuthenticationService, deployment *appsv1.Deployment) error {
+
+	install.ApplyDeploymentDefaults(deployment, "standard-authservice", authservice.Name)
+
+	install.ApplyInitContainerWithError(deployment, "keycloak-plugin", func(container *corev1.Container) error {
```

Ignoring the error will cause a warning in some IDEs. However there is a variant of this function `ApplyInitContainer` which does not require an error.

---

Fixed